### ISSUE TITLE
personal_newAccount JSON RPC calls always timeout after blockchain sync starts

### ISSUE THREAD

**Issue by BlameByte** [ethereum/go-ethereum#2993]
#### System information

Geth version: 1.4.10
OS & Version: Linux CentOS 7
Commit hash : release/1.4

#### Expected behaviour
For calls via the personal_newAccount to succeed and return a new address like what happens before blockchain sync starts.

#### Actual behaviour
As soon as it states "Block synchronisation started" it no longer accepts personal_newAccount requests despite having personal in the rpcapi param and the request simply is ignored and no error is returned.

Note: This still does not work even if we are fully synced (at the maximum height).

#### Steps to reproduce the behaviour
Enable the rpc using --rpc --rpcapi "eth,personal" and then try a request, if it is done before the block sync it will work, if it is during / after it will simply time out.

You can use --nodiscover --maxpeers 0 and you can always create new accounts; but then you are no longer in sync with the network.

#### Backtrace

Is there any way to keep access to the personal_newAccount("passphrase") calls so I can generate new addresses while still having an up to date node?

**Comment by karalabe** [comment]
Are you sure your machine is not overloaded and simply cannot generate the account? Creating a new account is an expensive operation. It requires 256MB memory and on a modern processor should take at least 1 sec, possibly more. If your machine is being hammered (e.g. during sync), it may take much longer to do the account crypto. Of course it should still succeed eventually. What machine are you running this on?

**Comment by BlameByte** [comment]
The instance I am using is fairly low spec: 1 CPU, 512MB ram (2GB swap), 20GB for os / swap and 100GB hdd for blockchain / storage.

It seems to fail even when it is fully synced and not using any CPU, there is little physical memory free while geth is running but there is plenty of swap on a separate drive. (1929004 free, 168144 used)

The rpc request times out and does not appear to process the request and still times out even when setting the timeout value to like 1 minute. 

Before Geth starts syncing the requests work as they should within a few seconds but then it seems as soon as it starts syncing a couple of blocks requests no longer work. Despite working before sync and even though it is fully synced and cpu usage drops to almost nothing (memory committed still remains high).

Is this due to the memory allocated during the sync loading the blockchain and it no longer has any left to create an address? Is swap unable to be used instead of physical ram for some reason (some swap is being used for syncing). 

Is there any way to check to see if a request succeeded, or pending requests?

**Comment by karalabe** [comment]
If you are hitting the swap space, it will probably slow down to an unusable crawl. Please reduce the database caching to something way less (e.g. `--cache=32`) so that even with caches and other memory usage you'll still have enough memory for the 256MB key operations. If you hit swap, say goodbye to responsiveness.

**Comment by BlameByte** [comment]
I tried again with an instance with 1GB ram and no swap as that was likely impacting performance, you are right it is definitely due to memory exhaustion.

Setting cache down didn't help much on either machine, it seems like it just continues to consume memory and then we no longer have enough to generate addresses (with 32mb cache we still use more than 800mb), and the only difference I have noticed is that the memory usage stays smaller but eventually goes up to the same regardless of caches.

Right now it is keeping a free 70mb and so creating an address is working perfectly but I will have to do some further testing to make sure that it will not run out of memory for address generation.

How much memory is considered adequate for a geth server and is 1GB too low? Also is a fast node a lot more memory intensive and would I require less if I were to sync from scratch without the --fast option?
